core/deployment/src/main/java/io/quarkus/deployment/configuration/MapConfigType.java

```diff
@@ -95,14 +95,39 @@ ResultHandle generateGetOrCreate(final BytecodeCreator body, final ResultHandle
     }
 
     void acceptConfigurationValueIntoLeaf(final LeafConfigType leafType, final NameIterator name, final SmallRyeConfig config) {
-        // leaf types directly into map values
-        throw Assert.unsupported();
+        if (leafType instanceof MapValueConfigType) {
```

Comment:
Why did you end up needing a separate config type here? I would have expected that the normal non-primitive leaf types would be usable.